expertsystem/data.py

```diff
@@ -288,13 +289,13 @@ def compute_isospin_projection(  # pylint: disable=too-many-arguments
 class ParticleCollection(abc.Mapping):
     """Safe, `dict`-like collection of `.Particle` instances."""
 
-    def __init__(
-        self, particles: Optional[Dict[str, Particle]] = None
-    ) -> None:
+    def __init__(self, particles: Optional[Iterable[Particle]] = None) -> None:
```

Comment:
I'm thinking again about the data structure of the `ParticleCollection` itself. Why can't it be a set again?

Reply:
It's mainly for the faster lookup. We assume most look up by name, so that's why we use a `dict` with names as keys. Otherwise you'd need to loop over the set somehow.

Reply:
True, thanks